File: Decorator/PHP/Decaf.php
<?php
  class Decaf extends Beverage {
    public function __construct() {
      parent::__construct();
      $this->description = " Decaf";
    }

    public function cost() {
      return 6.0;
    }

    public function getDescription() {
      return $this->description;
    }
  }

File: Decorator/Go/decorator/decaf.go
package decorator

type Decaf struct {
	*Coffee
}

func NewDecaf() *Decaf {
	return &Decaf{&Coffee{price: 2.5, description: "Decaf"}}
}

func (self *Decaf) Cost() float64 {
	return self.price
}

func (self *Decaf) GetDescription() string {
	return self.description
}
